ios/Plugin/HttpRequestHandler.swift

```diff
@@ -178,27 +173,25 @@ class HttpRequestHandler {
         let timeout = (connectTimeout ?? readTimeout ?? 600000.0) / 1000.0;
         request.setTimeout(timeout)
 
-        if isHttpMutate {
-            do {
-                guard let data = call.jsObjectRepresentation["data"]
-                else {
-                    throw CapacitorUrlRequest.CapacitorUrlRequestError.serializationError("Invalid [ data ] argument")
+        if let _ = call.options["data"] {
+            if let data = call.jsObjectRepresentation["data"] {
+                do {
+                    try request.setRequestBody(data)
+                } catch {
+                    // Explicitly reject if the http request body was not set successfully,
+                    // so as to not send a known malformed request, and to provide the developer with additional context.
+                    call.reject("Error", "REQUEST", error, [:])
+                    return
                 }
-                
-                try request.setRequestBody(data)
-            } catch {
-                // Explicitly reject if the http request body was not set successfully,
-                // so as to not send a known malformed request, and to provide the developer with additional context.
-                call.reject("Error", "REQUEST", error, [:])
-                return;
+            } else {
+                throw CapacitorUrlRequest.CapacitorUrlRequestError.serializationError("Invalid [ data ] argument")
```

Comment:
```suggestion
                call.reject("Error, "REQUEST", CapacitorUrlRequest.CapacitorUrlRequestError.serializationError("Invalid [ data ] argument"))
```

Something like this needs to be called since the error is no longer being caught by the `try` like before.

Reply:
Of course this is safe like this, but there is also a safety net in the Plugin.swift, where this error should be caught. As we implemented the fix, we also thought about directly rejecting here instead of throwing an error, so for us it is fine your suggested solution. I'll update the PR later on.